File: cpp/intersection_of_two_arrays.cpp
class Solution {

public:

    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {

        vector<int> res;

        int size1=nums1.size();

        int size2=nums2.size();

        if (size1==0 || size2==0) return res;

        sort(nums1.begin(), nums1.end());

        sort(nums2.begin(), nums2.end());

        

        int i=0, j=0;

        while (i<size1 && j<size2) {

            if (nums1[i]<nums2[j]) {

                i++;

            } else if (nums1[i]>nums2[j]) {

                j++;

            } else {

                int inter=nums1[i];

                res.push_back(inter);

                while (i<size1 && nums1[i]==inter) i++;

                while (j<size2 && nums2[j]==inter) j++;

            }

        }

        

        return res;

    }

};

2.
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        unordered_set<int> st1;
        set<int> resset;
        for (auto i:nums1) {
            st1.insert(i);
        }
        
        for (auto i:nums2) {
            if (st1.find(i)!=st1.end()) {
                resset.insert(i);
            }
        }
        
        for (auto i:resset) res.push_back(i);
        return res;
    }
};

File: java/intersection_of_two_arrays.java
public class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        List<Integer> inter=new ArrayList<Integer>();
        int n1=nums1.length;
        int n2=nums2.length;
        if (n1==0||n2==0) return new int[0];
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        
        for (int i=0, j=0; i<n1 && j<n2; ) {
            if (nums1[i]<nums2[j]) {
                i++;
            } else if (nums1[i]>nums2[j]) {
                j++;
            } else {
                int num=nums1[i];
                inter.add(num);
                while (i<n1 && nums1[i]==num) i++;
                while (j<n2 && nums2[j]==num) j++;
            }
        }
        
        int[] res=new int[inter.size()];
        for (int i=0; i<inter.size(); i++) res[i]=inter.get(i);
        return res;
    }
}
